### PR TITLE
Support for Zsh tab-completion

### PR DESCRIPTION
**Pull request by andrewnaguib**
## Motivation

Adding support for Zsh tab-completion

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebookresearch/hydra/blob/master/CONTRIBUTING.md)?

Yes

## Test Plan

The PR is still in a draft mode. Tests should be added beforehand developing the core function (following TDD). 

## Related Issues and PRs

(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

### CODE REVIEW

File: hydra/_internal/core_plugins/__init__.py [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 from .basic_launcher import BasicLauncher
 from .basic_sweeper import BasicSweeper
-from .bash_completion import BashCompletion
 
-__all__ = ["BasicLauncher", "BashCompletion", "BasicSweeper"]
+__all__ = ["BasicLauncher", "BasicSweeper"]
+
+from .completion import *
```
Comment (andrewnaguib) [inline hydra/_internal/core_plugins/__init__.py]: @omry Flake fails in this line. It's, originally, used to allow for `completion` plugins to be inspected directly from the `core_plugins`. I'm not sure what would be a workaround.

**Comment by andrewnaguib** [comment]
@omry For some reason the draft option didn't show up. ._.

File: website/docs/development/contributing.md [hunk 2]
```diff
@@ -26,7 +26,7 @@ conda activate hydra37
 ```
 From the source tree, install Hydra in development mode with the following command:
 ```
-pip install -e .[dev] -e .
+pip install -e '.[dev]' -e .
```
Comment (andrewnaguib) [inline website/docs/development/contributing.md]: I added the single quotes to properly install it for _both_ Bash and Zsh (it fails because of the `[]`).

Comment (omry) [reply website/docs/development/contributing.md]: roger.

File: hydra/_internal/core_plugins/__init__.py (hunk 1 above)
Comment (omry) [reply hydra/_internal/core_plugins/__init__.py]: Just change imports that are importing the bash completion and zsh completion to do it from the new module.

**Comment by omry** [comment]
Closing for now, please feel free to create a new PR when you want to continue working on this.

**Pull request closed without merging**